File: python/adt7410.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

import smbus

address = 0x48

block = smbus.SMBus(1).read_i2c_block_data(address, 0x00, 12)
temp = (block[0] << 8 | block[1]) >> 3
if temp >= 4096:
	temp -= 8192
print(temp/16.0, '℃')

File: ruby/adt7410.rb
#!/usr/bin/ruby
# -*- coding: utf-8 -*-

if $0 == __FILE__
	require_relative 'i2c'
	ADT7410 = 0x48
	DATA_SIZE = 12
	buf = i2c_read ADT7410, 0.chr, DATA_SIZE
	temp = (buf[0].ord << 8 | buf[1].ord) >> 3
	temp -= 8192 if temp & 0xf000 != 0
	printf "%f℃\n", temp/16.0
	buf.each_byte do |i|
		printf '%02x ', i.ord
	end
end
